### PR TITLE
[Feature]  Update coins balance when completing a task

### PR DESCRIPTION
**Pull request by daiHash**
## Description
Update coins balance when completing a task

## Screenshots
update-coin-balance

## Changes include

- [] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

## Checklist

- [x] I have tested this code

### CODE REVIEW

File: src/pages/DailyGardening/index.tsx [hunk 1]
```diff
@@ -86,8 +87,8 @@ export const DailyGardening = () => {
   const completeTaskHandler = useCallback(
     async (rule: Rule) => {
       if (rule._id) {
-        setLastClicked(rule._id)
-      };
+        setLastClicked(rule._id);
+      }
```
Comment (daiHash) [inline src/pages/DailyGardening/index.tsx:91]: Not sure why it was deleted in my pr sorry I ended up adding it again🙏 cc: @conalli

**Review by kome12 — APPROVED** [review]
LGTM!

File: src/pages/DailyGardening/index.tsx [hunk 2]
```diff
@@ -106,12 +107,23 @@ export const DailyGardening = () => {
...
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [userData]
   );
 
+  useEffect(() => {
+    if (completedTaskApi.status === "succeeded") {
+      const { balance: newCoinBalance } = completedTaskApi.response.user;
+      setUserData((data) => {
+        return { ...data, balance: newCoinBalance };
+      });
```
Comment (daiHash) [inline src/pages/DailyGardening/index.tsx:122]: updating global state coin balance

File: src/pages/DailyGardening/index.tsx [hunk 3]
```diff
@@ -163,27 +175,37 @@ export const DailyGardening = () => {
...
                   <Card variant="outlined" key={rule._id} color="background">
-                    <LoadingWrapper isLoading={lastClicked === rule._id && completedTaskApi.status === "loading"}>
-                    <Chip
-                      icon={
-                        isRuleCompleted(rule._id) ? <DoneIcon /> : <CloseIcon />
+                    <LoadingWrapper
+                      isLoading={
+                        lastClicked === rule._id &&
+                        completedTaskApi.status === "loading"
                       }
```
Comment (daiHash) [inline src/pages/DailyGardening/index.tsx:182]: Same as above ended up adding conall's changes again🙇‍♂️

**Pull request merged**